### ISSUE TITLE
Creating a new entry in a collection and saving the page redirects to a url containing "undefined" and getting a 404

### ISSUE THREAD

**Issue by Utibu** [statamic/cms#9283]
### Bug description

After updating our site to >4.38 we seem to have a bug when creating a new entry, independent of which collection we're using. Cannot find any reports of this bug here in the issues either.

The bug is behaving like this:

After I've filled filled in some content/title for a non-existing entry, I press save. What used to happen is that I got redirected to the entry and could continue editing it, but what instead happens now is that I get redirected to a URL containing "undefined", and I see a error message "Page Not Found".

### How to reproduce

1. Press on the collection (pages for example)
2. Create a new entry
3. Type in a title and click save
4. The URL should now be like _cp/collections/pages/entries/create/undefined?created=true_ and a "Page Not Found" should show.

### Environment

```yaml
Application Name: Laravel
Laravel Version: 10.40.0
PHP Version: 8.2.11
Composer Version: -
Environment: local
Debug Mode: ENABLED
URL: laravel.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / bugsnag
Mail: smtp
Queue: database
Session: file

Locales
Installed
LaravelLang\Actions\Plugin: 1.4.1
LaravelLang\Attributes\Plugin: 2.9.1
LaravelLang\HttpStatuses\Plugin: 3.8.0
LaravelLang\Lang\Plugin: 13.12.0
Locales Version: 1.9.0
Protected
Publisher Version: 15.0.2

Statamic
Addons: 2
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.42.1 PRO

Statamic Addons
mitydigital/statamic-two-factor: 1.1.4
statamic/eloquent-driver: 2.10.0

Statamic Eloquent Driver
Asset Containers: eloquent
Assets: eloquent
Blueprints: file
Collection Trees: file
Collections: file
Entries: eloquent
Forms: eloquent
Global Sets: file
Global Variables: eloquent
Navigation Trees: eloquent
Navigations: eloquent
Revisions: eloquent
Taxonomies: eloquent
Terms: eloquent
```

### Installation

Existing Laravel app

### Antlers Parser

None

**Comment by duncanmcclean** [comment]
We had this issue a few months ago, which was fixed in #9095.

Can you try clearing your views cache? `php artisan view:clear`

**Comment by Utibu** [comment]
> Have you published any of Statamic's views? (if you have, they'll be in `resources/views/vendor/statamic`)

Only one from what I can see, partials/favicon.antlers.html.

**Comment by duncanmcclean** [comment]
Can you try running `php artisan statamic:install` (it'll ensure you have the latest JS for the Control Panel) and see if that helps at all?

Do you have anything else that "overrides" default Statamic behaviour, like custom controllers?

**Comment by Utibu** [comment]
Hi again! I think I found a breakthrough on this issue.

I tried running statamic:install with no success, so I tried to just brute force remove everything with a possible connection to Statamic we've made custom. We've got some user groups eloquent stuff, and some custom css/js/fieldtypes. But none of them solved the problem. But then I removed the line `JsonResource::withoutWrapping();` in our AppServiceProvider, and it suddenly worked!

I've seen some other issues now regarding that line of code, #7068 and #303. From our experience that line of code is pretty well used in Laravel projects, so this should probably be fixed or at least mentioned in the documentation as a potential problem maybe?

I guess a workaround could be to only append it on routes not a part of the control panel, but I haven't got that far yet.
